### PR TITLE
[MPMD-325] Could not find class due to IncompatibleClassChangeError warning

### PR DESCRIPTION
**Pull request by adangel**
- Fixes [MPMD-325](https://issues.apache.org/jira/browse/MPMD-325)
- Fixes [MPMD-326](https://issues.apache.org/jira/browse/MPMD-326)

Following this checklist to help us incorporate your
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MPMD) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue.  Your pull request should address just this issue, without
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MPMD-XXX] - Subject of the JIRA Ticket`,
       where you replace `MPMD-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/it/MPMD-270-325-JDK11/pom.xml [hunk 1]
```diff
@@ -28,6 +28,7 @@ under the License.
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <java.version>11</java.version>
```
Comment (michael-o) [inline src/it/MPMD-270-325-JDK11/pom.xml:31]: 11 here?

File: src/it/MPMD-280-JDK12/invoker.properties [hunk 2]
```diff
@@ -16,4 +16,5 @@
 # under the License.
 
 invoker.java.version = 12+
```
Comment (michael-o) [inline src/it/MPMD-280-JDK12/invoker.properties:18]: But here 12?

Comment (adangel) [reply src/it/MPMD-280-JDK12/invoker.properties:18]: that's correct - these are different test cases:

* src/it/MPMD-270-325-JDK11/pom.xml verifies java11 usage
* src/it/MPMD-280-JDK12/invoker.properties verifies java12 usage

File: pom.xml [hunk 3]
```diff
@@ -102,6 +102,13 @@ under the License.
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
       <version>${mavenVersion}</version>
+      <exclusions>
```
Comment (adangel) [inline pom.xml:105]: this exclusion is actually not necessary anymore after upgrading to maven-core to 3.1.1

**Review by slachiewicz — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #295

**Pull request merged**